service/matching/pollerHistory.go

```diff
@@ -30,7 +30,7 @@ import (
 
 const (
 	pollerHistoryInitSize    = 0
-	pollerHistoryInitMaxSize = 1000
+	pollerHistoryInitMaxSize = 5000
```

Comment:
Yes, this is per tasklist. 5000 should be enough for most cases, if we will see 5000 pollers per tasklist, it might be a good indicator to turn on tasklist sharding